The failing test that this PR replaces was always passing. The construction of the `Radix` was not following the same flow as the Ruby code, which has been corrected.

```console
PS C:\Users\rjl\dev\artichoke\artichoke> cargo run --bin airb
   Compiling artichoke-backend v0.1.0 (C:\Users\rjl\dev\artichoke\artichoke\artichoke-backend)
   Compiling artichoke v0.1.0-pre.0 (C:\Users\rjl\dev\artichoke\artichoke)
    Finished dev [unoptimized + debuginfo] target(s) in 5.08s
     Running `target\debug\airb.exe`
artichoke 0.1.0-pre.0 (2021-08-16 revision 4624) [x86_64-pc-windows-msvc]
[rustc 1.54.0 (a178d0322 2021-07-26) on x86_64-pc-windows-msvc]
>>>
=> nil
>>> Integer("0z", 12000)
Traceback (most recent call last):
        2: from (airb):1
        1: from (eval):92:in Integer
ArgumentError (invalid radix 12000)
>>>
```

--- a/artichoke-backend/src/extn/core/kernel/integer.rs
+++ b/artichoke-backend/src/extn/core/kernel/integer.rs
@@ -73,10 +73,8 @@ impl TryConvertMut<Option<Value>, Option<Radix>> for Artichoke {
                     .checked_neg()
                     .ok_or_else(|| ArgumentError::with_message("invalid radix"))?;
                 match u32::try_from(num) {
-                    // Safety `10` is a valid radix since it is between 2 and 36.
-                    //
                     // See https://github.com/ruby/ruby/blob/v2_6_3/bignum.c#L4106-L4110
-                    Ok(1) => return unsafe { Ok(Some(Radix::new_unchecked(10))) },
+                    Ok(1) => return Ok(Some(Radix::default())),
                     Ok(radix) => radix,
                     Err(_) => {
                         let mut message = String::from("invalid radix ");

